### PR TITLE
ENH: Enable hyperparameter search with AML SDK v2

### PR DESCRIPTION
**Pull request by mebristo**
Enable hyperparameter search with AML SDK v2

### CODE REVIEW

File: hi-ml-azure/src/health_azure/himl.py [hunk 1]
```diff
@@ -429,20 +478,61 @@ def submit_run_v2(workspace: Optional[Workspace],
...
+
+        job_to_submit = command_job.sweep(
+            compute=compute_target,  # AML docs suggest setting this here although already passed to command
+            **hyperparam_args
+        )
+
+        # AML docs state to reset certain properties here which aren't picked up from the
+        # underlying command such as experiment name and max_total_trials
+        job_to_submit.experiment_name = experiment_name
+        job_to_submit.set_limits(max_total_trials=hyperparam_args.get("max_total_trials", None))
```
Comment (ant0nsc) [inline hi-ml-azure/src/health_azure/himl.py:518]: can you define a constant for "max_total_trials"? It is used at multiple places.

**Review by ant0nsc — APPROVED** [review]
Is it clear in the documentation what people should do when they want to do custom hyperdrive (ie, not any of the predefined one for crossval or seeds)?

File: hi-ml-azure/src/health_azure/himl.py [hunk 2]
```diff
@@ -429,20 +478,61 @@ def submit_run_v2(workspace: Optional[Workspace],
...
+            code=str(snapshot_root_directory),
+            command=cmd,
+            inputs=input_datasets_v2,
+            outputs=output_datasets_v2,
+            environment=environment.name + "@latest",
+            compute=compute_target,
+            experiment_name=experiment_name,
+            environment_variables={
+                "JOB_EXECUTION_MODE": "Basic",
+                "AZUREML_COMPUTE_USE_COMMON_RUNTIME": "true"
```
Comment (ant0nsc) [inline hi-ml-azure/src/health_azure/himl.py:500]: Do we really need that AZUREML_COMPUTE_USE_COMMON_RUNTIME anymore? here and in line 531

File: hi-ml/src/health_ml/lightning_container.py [hunk 3]
```diff
@@ -183,6 +183,21 @@ def get_hyperdrive_config(self) -> Optional[HyperDriveConfig]:
...
         return None
 
+    def get_hyperparam_args(self) -> Optional[Dict[str, Any]]:
+        """
+        Returns a dictionary of hyperparameter search arguments that will be passed to an AML v2 command to
+        enable either hyperparameter tuning,  cross validation, or running with different seeds.
+
+        :return: A dictionary of hyperparameter search arguments and values.
+        """
+        if self.hyperdrive:
```
Comment (kenza-bouzid) [inline hi-ml/src/health_ml/lightning_container.py:193]: hyperdrive is not available with the new sdk only right? How about checking here that strictly_aml_v1 is False and suggesting to switch to the sdk v1 for hyperdrive runs?

**Review by kenza-bouzid — APPROVED** [review]

**Comment by mebristo** [comment]
@ant0nsc, @kenza-bouzid  Thanks for the reviews! I have added the ability to run hyperparameter search and included an example of how to do so in "hyperparemeter_search.md". I think this should address your remaining concerns?

**Pull request merged**